src/server/create-renderer.js

```diff
@@ -10,9 +10,18 @@ export function createRenderer ({
   return {
     renderToString (component) {
       let result = ''
+      let stackDepth = 0
       render(component, (str, next) => {
         result += str
-        next && next()
+        if (next) {
+          if (stackDepth >= 1000) {
```

Comment:
Let's export `MAX_STACK_DEPTH` as a const here and import it in `render-stream.js`.